pkg/KubeArmorOperator/internal/controller/cluster.go

```diff
@@ -103,6 +104,136 @@ func NewClusterWatcher(client *kubernetes.Clientset, log *zap.SugaredLogger, ext
 	}
 }
 
+func extractVolumeFromMessage(message string) (string, bool) {
+	// find volume name between quotes after "volume"
+	// Message: MountVolume.SetUp failed for volume \"notexists-path\"
+	re := regexp.MustCompile(`volume\s*\"([^\"]+)\"`)
+	matches := re.FindStringSubmatch(message)
+
+	if len(matches) > 1 {
+		return matches[1], true
+	}
+	return "", false
+}
+
+func extractPathFromMessage(message string) (string, bool) {
+	// find mount path between quotes after "volume"
+	// Message: failed to mkdir \"/etc/apparmor.d/\": mkdir /etc/apparmor.d/: read-only file system
+	re := regexp.MustCompile(`mkdir\s+\"([^\"]+)\"`)
+	matches := re.FindStringSubmatch(message)
+
+	if len(matches) > 1 {
+		return matches[1], true
+	}
+	return "", false
+}
+
+func (clusterWatcher *ClusterWatcher) checkJobStatus(job, runtime, nodename string) {
+	defer func() {
+		clusterWatcher.Log.Infof("checkJobStatus completed for job: %s", job)
+	}()
+
```

Comment:
Let's use a 5 minute timeout here just in case this does not exit

Reply:
I believe when it gets success response then the job resource has been created persisted in etcd. can you think of any edge case here and we should have a delay in that case.

Reply:
I mean inside the infinite for loop not in defer function.

Reply:
I got that, 
I just wanted to mention It should exit in all the cases, In the infinite loop if are you suggesting a delay I think we can have that but 5 minutes is long as in valid cases it will add a long delay.